cfme/tests/integration/test_cfme_auth.py

```diff
@@ -197,9 +186,82 @@ def _group_cleanup(group):
             request.addfinalizer(lambda: _group_cleanup(group))
             navigate_to(appliance.server, 'LoggedIn', wait_for_view=True)
             assert appliance.server.current_full_name() == user.name
-            assert group.description.lower() in [name.lower() for
-                                                 name in appliance.server.group_names()]
+            assert group.description.lower() in [name.lower()
+                                                 for name in appliance.server.group_names()]
 
     appliance.server.login_admin()
     for user, _ in user_group_tuples:
         assert user.exists
+        request.addfinalizer(user.delete)
+
+
+@pytest.fixture(scope='function')
+def local_group(appliance):
+    """Helper method to check for existance of a group and delete if need be"""
+    group_collection = appliance.collections.groups
+    group_args = {
+        'description': 'test-group-{}'.format(gen_alphanumeric(length=5)),
+        'role': 'EvmRole-desktop'}
+    group = group_collection.create(**group_args)  # gets re-instantiated in create
+    assert group.exists
+    yield group
+
+    try:
+        group.delete()
```

Comment:
Required: Let's add a check for existence instead try block:
    if group.exists:
        group.delete()